File: 2/2/pdoo/prac/ruby/enemystarship.rb
#encoding:utf-8

require_relative "./lib/EnemyToUI.rb"
require_relative "./shotresult.rb"

module Deepspace
  # Nave enemiga
  class EnemyStarship

    attr_reader :name, :ammoPower, :shieldPower, :loot, :damage

    def initialize(n,a,s,l,d)
      @name = n
      @ammoPower = a
      @shieldPower = s
      @loot = l
      @damage = d
    end

    # Constructor por copia
    def self.newCopy(e)
      EnemyStarship.new(e.name,e.ammoPower,e.shieldPower,e.loot,e.damage)
    end

    # Análogo al consultor del atributo 'ammoPower'
    def fire
      @ammoPower
    end

    # Análogo al consultor del atributo 'shieldPower'
    def protection
      @shieldPower
    end
    
    # Determina si la nave aguanta un disparo
    def recieveShot(shot)
      if @shieldPower<shot
        return ShotResult::DONOTRESIST
      else
        return ShotResult::RESIST
      end
    end

    def getUIVersion
      EnemyToUI.new(self)
    end

    def to_s
      "name = #{@name}\n
      ammoPower = #{@ammoPower}\n
      shieldPower = #{@shieldPower}\n
      loot = #{@loot.to_s}\n
      damage = #{@damage.to_s}"
    end
  end
end

File: 2/2/pdoo/prac/java/deepspace/src/Deepspace/EnemyStarShip.java
/**
 * @author Manuel Gachs Ballegeer
 */
package Deepspace;

/**
 * @brief Clase nave enemiga
 */
class EnemyStarShip {
    private Damage damage;
    private Loot loot;
    private String name;
    private float ammoPower;
    private float shieldPower;
    
    EnemyStarShip(String n, float a, float s,Loot l,Damage d) {
        damage = d;
        loot = l;
        name = n;
        ammoPower = a;
        shieldPower = s;
    }
    
    EnemyStarShip(EnemyStarShip e) {
        damage = e.damage;
        loot = e.loot;
        name = e.name;
        ammoPower = e.ammoPower;
        shieldPower = e.shieldPower;
    }
    
    EnemyToUI getUIversion() { return new EnemyToUI(this); }
    
    public float fire() { return ammoPower; }
    
    public float getAmmoPower() { return ammoPower; }
    
    public Damage getDamage() { return damage;  }
    
    public Loot getLoot() { return loot; }
    
    public String getName() { return name; }
    
    public float getShieldPower() { return shieldPower; }
    
    public float protection() { return shieldPower; }
    
    public ShotResult recieveShot(float shot) {
        if (shieldPower<shot)
            return ShotResult.DONOTRESIST;
        else
            return ShotResult.RESIST;
    }
}
